<launch>
<!-- README: IF YOU WANT TO USE THIS LAUNCH FILE YOU SHOULD DO:
        git checkout simulador 
        INTO THE COLA2_TURBOT PACKAGE -->
    <!-- Params -->
    <arg name="place" default="bendinat.yaml"/>
    <arg name="robot1_name" default="turbot"/>
    <arg name="robot2_name" default="turbot2"/>
    
    <!-- Flags -->
    <arg name="cola2_nav" default="false"/>
    <arg name="enableNav" default="true"/>
    <arg name="enableUSBLloc" default="false"/>
    
    <!-- Interfaces -->
    <arg name="enableIqua" default="false"/>
    <arg name="enableIqua2" default="false"/>

    <arg name="enableMultimaster" default="false"/>

    <arg name="simulation_rate" default="50"/>
    <arg name="graphics_quality" default="low"/>

    <!-- robot -->
    <arg name="robot1" default="true"/>
    <arg name="robot2" default="true"/>


    <!-- ROBOT 1 -->
    <group if="$(arg robot1)">
        
        <include file="$(find cola2_turbot)/launch/robot.launch">
            <arg name="place" value="$(arg place)"/>
            <arg name="robot_name" value="$(arg robot1_name)"/>
            <arg name="sim" value="true"/>
            <arg name="simGPS" value="true"/>
            <arg name="enableComputerLogger" default="true"/>
            <arg name="cola2_nav" value="$(arg cola2_nav)"/>
            <arg name="enableNav" default="$(arg enableNav)"/>
            <arg name="enableSafety" value="false"/>
            <arg name="enableThrusters" value="false"/>
            <arg name="openLoopTeleoperation" value="false"/>
            <arg name="enableUSBLloc" value="$(arg enableUSBLloc)"/>
            <arg name="enableCamera" value="false"/>
            <arg name="enableModem" value="false"/>
            <arg name="enableRelayBoard" value="false"/>
            <arg name="enableIqua" value="$(arg enableIqua)"/>
            <arg name="enableMultimaster" value="$(arg enableMultimaster)"/>
            <arg name="enableEchosounder" value="false"/>
            <arg name="world2ned" value="false"/>
        </include>

        <!-- Internal Sensors -->
        <group ns="$(arg robot1_name)">
            <rosparam command="load" file="$(find cola2_turbot)/config/sim_internal_sensors.yaml"/>
            <node name="sim_internal_sensors" pkg="cola2_sim" type="sim_internal_sensors.py" output="screen"/>
            <!-- <node name="sim_pressure_sensor" pkg="cola2_turbot" type="depth_sensor_publisher.py" output="screen"/> -->
        </group>

    </group>

    <!-- ROBOT 2 -->
    <group if="$(arg robot2)">
        <include file="$(find cola2_turbot)/launch/robot.launch">
            <arg name="place" value="$(arg place)"/>
            <arg name="robot_name" value="$(arg robot2_name)"/>
            <arg name="sim" value="true"/>
            <arg name="simGPS" value="true"/>
            <arg name="enableComputerLogger" default="true"/>
            <arg name="cola2_nav" value="$(arg cola2_nav)"/>
            <arg name="enableSafety" value="false"/>
            <arg name="enableThrusters" value="false"/>
            <arg name="openLoopTeleoperation" value="false"/>
            <arg name="enableUSBLloc" value="$(arg enableUSBLloc)"/>
            <arg name="enableCamera" value="false"/>
            <arg name="enableModem" value="false"/>
            <arg name="enableRelayBoard" value="false"/>
            <arg name="enableIqua" value="$(arg enableIqua2)"/>
            <arg name="enableMultimaster" value="$(arg enableMultimaster)"/>
            <arg name="enableEchosounder" value="false"/>
            <arg name="world2ned" value="false"/>
        </include>

        <!-- SIMULATE SENSORS and ACTUATORS -->
        <group ns="$(arg robot2_name)">
            <!-- Internal Sensors -->
            <rosparam command="load" file="$(find cola2_turbot)/config/sim_internal_sensors.yaml"/>
            <node name="sim_internal_sensors" pkg="cola2_sim" type="sim_internal_sensors.py" output="screen"/>
            <!-- <node name="sim_pressure_sensor" pkg="cola2_turbot" type="depth_sensor_publisher.py" output="screen"/> -->
        </group>

        <group ns="$(arg robot2_name)">
            <rosparam command="load" file="$(find cola2_turbot)/config/simulation/config/frames2.yaml"/>
            <rosparam command="load" file="$(find cola2_turbot)/config/simulation/config/robot_localization2.yaml" ns="navigator"/>
        </group>
    </group>

    <!-- Run Simulator -->
    <!-- <include file="$(find cola2_turbot)/launch/modules/stonefish.launch">
        <arg name="simulation_data" value="$(find cola2_turbot)/config/simulation"/>
        <arg name="scenario_description" value="$(find multi_robot_system)/config/simulation/multi_robot_scenario.scn"/>
        <arg name="simulation_rate" value="$(arg simulation_rate)"/>
        <arg name="graphics_resolution" value="50 50"/>
        <arg name="graphics_quality" value="$(arg graphics_quality)"/>
    </include> -->

    <!-- <include file="$(find cola2_turbot)/launch/modules/stonefish_nogpu.launch">
        <arg name="simulation_data" value="$(find cola2_turbot)/config/simulation"/>
        <arg name="scenario_description" value="$(find multi_robot_system)/config/simulation/multi_robot_scenario_nogpu.scn"/>
        <arg name="simulation_rate" value="$(arg simulation_rate)"/>
    </include> -->

</launch>